src/Core/Domain/Cart/ValueObject/CartId.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartConstraintException;
+
+/**
+ * Cart identity
+ */
+class CartId
+{
+    /**
+     * @var int
+     */
+    private $cartId;
+
+    /**
+     * @param int $cartId
+     *
+     * @throws CartConstraintException if cart id is not valid.
+     */
+    public function __construct($cartId)
```

Comment:
@matks I'm thinking about static factories. Wdyt if we introduce factories like `CartId::fromString()` or `CartId::fromInt()`? We often need to cast argument to `int` (e.g. `new CartId((int) $cartIdWithStringType)`) so instead we would have `CartId::fromString($cartIdWithStringType)`.

Reply:
👍very good idea

Although I'm scared of usecases where `CartId::fromString()` is needed, why does the code have need for string cart IDs 😄?

Reply:
Hmm, if you get data from database or query parameter, it usually comes in `string` type. As you can see in code, we have a lot of casting at the moment, so I think it could make sense, but I don't think it's a must have.